src/Sylius/Behat/Context/Setup/CatalogPromotionContext.php

```diff
@@ -116,19 +116,24 @@ public function itWillReducePrice(CatalogPromotionInterface $catalogPromotion, f
     }
 
     /**
-     * @Given /^there is a catalog promotion with "([^"]*)" name that applies on ("[^"]+" variant) and reduces price by ("[^"]+")$/
+     * @Given /^there is a catalog promotion "([^"]*)" that reduces price by ("[^"]+") and applies on ("[^"]+" variant) and ("[^"]+" variant)$/
+     * @Given /^there is a catalog promotion "([^"]*)" that reduces price by ("[^"]+") and applies on ("[^"]+" variant)$/
      */
-    public function thereIsACatalogPromotionWithNameThatAppliesOnVariantAndReducesPriceBy(
+    public function thereIsACatalogPromotionThatReducesPriceByAndAppliesOn(
         string $name,
-        ProductVariantInterface $variant,
-        float $discount
+        float $discount,
+        ProductVariantInterface ...$variants
     ): void {
         $catalogPromotion = $this->createCatalogPromotion($name);
+        $variantCodes = [];
+        foreach ($variants as $variant) {
+            $variantCodes[] = $variant->getCode();
+        }
 
         /** @var CatalogPromotionRuleInterface $catalogPromotionRule */
         $catalogPromotionRule = $this->catalogPromotionRuleFactory->createNew();
         $catalogPromotionRule->setType(CatalogPromotionRuleInterface::TYPE_FOR_VARIANTS);
-        $catalogPromotionRule->setConfiguration(['variants' => [$variant->getCode()]]);
+        $catalogPromotionRule->setConfiguration(['variants' => array_values($variantCodes)]);
```

Comment:
```suggestion
        $catalogPromotionRule->setConfiguration(['variants' => $variantCodes]);
```
should be enough 🤔

Reply:
then it creates something like:  'variants' => 0: 'code1' , 1:'code2' that is why I used array_values